### DISCUSSION TITLE
ElevenLabs Voices?

### DISCUSSION THREAD

**Discussion by dangerpotter** [open-webui/open-webui#3379] [Q&A] [answered]
Is it possible to use an ElevenLabs voice for TTS? If so, how do I get it to work? In the audio settings I've toggled the TTS to OpenAI and then tried to use the ElevenLabs URL and API with no luck.

**Comment by rodrigomartinsvmc** [comment, marked as answer]
Hello, the new version of the webui already has EllevenLabs native to synthesize the voice, just select and insert the token, voice ID and model.

**Comment by microcoder-py** [comment]
Can you specify the settings for this

**Reply by nickfixit** [reply]
This does not work like it used to. Noatter what you set. I had this working a few patches ago with just simply changing the API url and then tweaking he settings. Now it doesnt with whatever combination I try.

**Reply by Sriharan-S** [reply]
The below Settings in OpenWebUI works perfectly.
Please let me know if it worked for you.

TTS Model: ```eleven_multilingual_v2```
TTS Voice: ```voice_id```

Voice IDs:
| Voice Name | Voice ID                  |
|------------|---------------------------|
| Aria       | 9BWtsMINqrJLrRacOk9x      |
| Roger      | CwhRBWXzGAHq8TQ4Fs17      |
| Sarah      | EXAVITQu4vr4xnSDxMaL      |
| Laura      | FGY2WhTYpPnrIDTdsKH5      |
| Charlie    | IKne3meq5aSn9XLyUdCD      |
| George     | JBFqnCBsd6RMkjVDRZzb      |
| Callum     | N2lVS1w4EtoT3dr4eOWO      |
| River      | SAz9YHcvj6GT2YYXdXww      |
| Liam       | TX3LPaxmHKxFdv7VOQHJ      |
| Charlotte  | XB0fDUnXU5powFXDhCwa      |
| Alice      | Xb7hH8MSUJpSbSDYk0k2      |
| Matilda    | XrExE9yKIg1WjnnlVkGX      |
| Will       | bIHbv24MWmeRgasZH58o      |
| Jessica    | cgSgspJ2msm6clMCkdW9      |
| Eric       | cjVigY5qzO86Huf0OWal      |
| Chris      | iP95p4xoKVk53GoZ742B      |
| Brian      | nPczCjzI2devNBz1zQrb      |
| Daniel     | onwK4e9ZLuTAKqWW03F9      |
| Lily       | pFZP5JQG7iQjIQuC4Bku      |
| Bill       | pqHfZKP75CvOlQylNhV4      |

**Reply by gelbzucht** [reply]
Did anyone manage to use the id of a custom voice as well? The regular ones work for me, however I don't get responses for custom voices.

**Reply by SudoZeus-Code** [reply]
Was the custom voice ever resolved? I'm trying to use a custom voice I generated on elevenlabs. When i try to use it I'm getting "Invalid voice id" error.  I tried switching between Multilingual and Turbo but no luck.

**Reply by microcoder-py** [reply]
@SudoZeus-Code I ended up directly modifying the codebase to achieve it, but yes I did achieve it

**Reply by PorterRWS** [reply]
@microcoder-py What did you have to modify, and have you submitted a bug report?

**Reply by SkylerGhostly** [reply]
@microcoder-py I'd also like to know this, as well, please!

**Reply by stefanneubig** [reply]
I've got the same problem. How did you fix it?

**Reply by microcoder-py** [reply]
@SkylerGhostly  @stefanneubig @PorterRWS I modified this function by hardcoding it to my requirements https://github.com/open-webui/open-webui/blob/0cef844168e97b70de2abee4c076cc30ffec6193/backend/open_webui/routers/audio.py#L281

**Reply by microcoder-py** [reply]
Cannot share the exact code since it is custom

But I had to hardcode this
